tests/valgrind/Makefile

```diff
@@ -18,8 +18,8 @@ all:
 clean:
 	rm -f ${PROG} ${SRCS:.c=.o}
 
-${PROG}:${SRCS:.c=.o}
-	${CC} -o ${PROG} ${SRCS:.c=.o} ${LDFLAGS} ${LDADD_EXTRA} ${LDADD_REQ} ${LDADD_POSIX}
+${PROG}:${SRCS:.c=.o} ${LIBALL}
+	${CC} -o ${PROG} ${SRCS:.c=.o} ${LIBALL} ${LDFLAGS} ${LDADD_EXTRA} ${LDADD_REQ} ${LDADD_POSIX}
```

Comment:
These changes are not desired, but end up being harmless, because we haven't defined ${LIBALL} in this `Makefile`. 
 The reason they appeared: `${LIBALL}` is now part of `Makefile.prog`.  Due to the `NOLIBALL = 1` in `Makefile.BSD`, we haven't defined that variable, but we're still copying `Makefile.prog` verbatim.

Technically, things would get confused if somebody has a `LIBALL` environment variable, but I think that's sufficiently unlikely that it's not worth complicating the build to strip it from this one test directory.

Reply:
I think I changed my mind -- I'll see about removing these instances of `${LIBALL}`.